presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -277,12 +283,6 @@ public ColumnHandle getUpdateRowIdColumnHandle(ConnectorSession session, Connect
         return primitiveIcebergColumnHandle(0, "$row_id", BIGINT, Optional.empty());
     }
 
-    @Override
-    public ConnectorTableHandle beginDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
-    {
-        throw new PrestoException(NOT_SUPPORTED, "This connector only supports delete where one or more partitions are deleted entirely");
-    }
-
```

Comment:
Quick question: As this PR turns to focus on supporting `TRUNCATE`, do we still need to remove this function?

Reply:
This method is incorrectly implemented and misdirects users. If someone were to query the connector `DELETE from <table> WHERE <predicate>;` the error would say "This connector only supports delete where one or more partitions are deleted entirely".

However, that statement is incorrect and misdirects users as the connector never supported deleting partitions. This should be removed until the support is added.